### PR TITLE
SDK-1757: Task 2. Connect S3 buckets

### PR DESCRIPTION
**Pull request by VladislavStrelchenko**
Ticket: https://spryker.atlassian.net/browse/SDK-1794

### CODE REVIEW

File: src/FileStorage/BucketFileStorage.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        $credentials = new Credentials(
+            $this->config->getFileBucketCredentialsKey(),
+            $this->config->getFileBucketCredentialsSecret(),
+        );
+
+        return new S3Client([
+            static::VERSION_KEY => static::LATEST_KEY,
+            static::REGION_KEY => $this->config->getFileBucketRegion(),
+            static::CREDENTIALS_KEY => $credentials,
+        ]);
```
Comment (DmytroKlymanSpryker) [inline src/FileStorage/BucketFileStorage.php:138]: On my opinion we should use a factory for that. Factory should set all this data to the client and return the object

Comment (DmytroKlymanSpryker) [reply src/FileStorage/BucketFileStorage.php:138]: Ok, as we've discussed, it is clear now to me why you've put it here.

File: src/Manifest/FileBucketManifestReader.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    /**
+     * @param int $releaseGroupId
+     *
+     * @return array<string, array<string, array<string>>>
+     */
+    public function readManifests(int $releaseGroupId): array
+    {
+        $fileContent = $this->bucketFileStorage->getFile($this->getObjectKey($releaseGroupId));
```
Comment (DmytroKlymanSpryker) [inline src/Manifest/FileBucketManifestReader.php:38]: AWS SDK has method `doesObjectExist` just if you need to check if the object exists in bucket.

**Review by DmytroKlymanSpryker — APPROVED** [review]

File: src/Manifest/FileBucketManifestReader.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/Manifest/FileBucketManifestReader.php:38]: ++
Thanks 
Added uses for the method

File: src/FileStorage/BucketFileStorage.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    /**
+     * @param string $filePath
+     *
+     * @return string|null
+     */
+    public function getFile(string $filePath): ?string
+    {
+        $client = $this->getS3Client();
```
Comment (pavelmaksimov25) [inline src/FileStorage/BucketFileStorage.php:84]: can be moved down, after the bucket name check

File: src/FileStorage/BucketFileStorage.php [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    /**
+     * @var string
+     */
+    protected const CONTENT_LENGTH_KEY = 'ContentLength';
+
+    /**
+     * @var \Aws\S3\S3Client $s3Client
+     */
+    protected ?S3Client $s3Client = null;
```
Comment (pavelmaksimov25) [inline src/FileStorage/BucketFileStorage.php:62]: I'd recommend extracting s3 client to the adapter class along with the `getS3Client` and `validateCredentials` methods.

File: src/FileStorage/BucketFileStorage.php (hunk 3 above)
Comment (VladislavStrelchenko) [reply src/FileStorage/BucketFileStorage.php:84]: in this case, the public method will be under the protected one - the situation is restricted by your clean-code rules

File: src/FileStorage/BucketFileStorage.php (hunk 4 above)
Comment (VladislavStrelchenko) [reply src/FileStorage/BucketFileStorage.php:62]: we will need to rewrite off method of the file if the S3Client changes - as a result the class is a kind of adapter 
is it suitable to rename the class to `BucketFileStorageAdapter`?

Comment (pavelmaksimov25) [reply src/FileStorage/BucketFileStorage.php:62]: yes. I didn't get it at the first place.

**Review by pavelmaksimov25 — APPROVED** [review]

**Pull request merged**